pkg/apis/cyclone/v1alpha1/workflow_trigger.go

```diff
@@ -0,0 +1,43 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// WorkflowTrigger describes trigger of an workflow, time schedule and webhook supported.
+type WorkflowTrigger struct {
+	// Metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+	// Metadata for the particular object, including name, namespace, labels, etc
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of this workflow trigger
+	Spec WorkflowTriggerSpec `json:"spec"`
+	// Status of this trigger
+	Status WorkflowTriggerStatus `json:"status"`
+}
+
+// Type of workflow trigger
+type TriggerType string
+
+const (
+	ScheduledTrigger = "Schedule"
+	WebhookTrigger   = "Webhook"
+)
+
+// WorkflowTriggerSpec defines workflow trigger definition.
+type WorkflowTriggerSpec struct {
+	// Name of the workflow to trigger
+	WorkflowRef string `json:"workflowRef"`
```

Comment:
Use objectreference here

```go
// ObjectReference contains enough information to let you inspect or modify the referred object.
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
type ObjectReference struct {
	// +optional
	Kind string
	// +optional
	Namespace string
	// +optional
	Name string
	// +optional
	UID types.UID
	// +optional
	APIVersion string
	// +optional
	ResourceVersion string

	// Optional. If referring to a piece of an object instead of an entire object, this string
	// should contain information to identify the sub-object. For example, if the object
	// reference is to a container within a pod, this would take on a value like:
	// "spec.containers{name}" (where "name" refers to the name of the container that triggered
	// the event) or if no container name is specified "spec.containers[2]" (container with
	// index 2 in this pod). This syntax is chosen only to have some well-defined way of
	// referencing a part of an object.
	// TODO: this design is not final and this field is subject to change in the future.
	// +optional
	FieldPath string
}
```